ocaml/xenopsd/xc/device.ml

```diff
@@ -3833,6 +3840,14 @@ module Dm = struct
                       crash_reason
             )
         )
+    ) ;
+    (* CP-46917 Wait for vgpu/demu to be running. Timeout needs to be
+       long enough for migrations to complete *)
```

Comment:
This function is called as part of the `VM_create_device_model` op that is part of the `VM_restore_devices` op. In the receiving host of a live migration, the `VM_receive_memory` op is executed, which runs `VM_restore_devices` after `VM_restore` and (just) before `VM_unpause` (after Synchronisation Point 3). See step 14 in [this diagram](https://xapi-project.github.io/new-docs/xenopsd/walkthroughs/live-migration/index.html). At this point, DEMU has already finished restoring the data from the sending host, so it won't take very long to get to the "running" state anymore.

Resume should be similar.

Reply:
To be safe, I restored the timeout to the value we had before but happy to adjust it down. The first timeout I've set to 60s because we should see "initialising" almost immediately.